var/spack/repos/builtin/packages/atk/package.py

```diff
@@ -41,6 +41,9 @@ class Atk(Package):
 
     depends_on('meson', type='build', when='@2.28.0:')
     depends_on('glib')
+    # FIXME: python3-dependency should only apply to 2.28.1 #9099
```

Comment:
The problem is slightly different - could you rephrase:

```
# FIXME: this constraint exists because of the meson dependency.
# It should not be required to specify it here, but "spack spec atk" will fail without it. 
# See: #2632
````

If you wanted the python3 dependency to only apply for `@2.28.1` or higher, you could rewrite the `depends_on` as follows:

```
depends_on('python@3:', when='@2.28.0:')
```

But the main problem is that you shouldn't have to specify that in the first place (and to be clear I'm not asking you to change the `depends_on` line, just the comment).

Reply:
Regarding

> See: #2632

The problem you encountered here comes up in many forms, #2632 is one of the original issues of this form.

Reply:
> The problem is slightly different - could you rephrase:

Done

Reply:
> If you wanted the python3 dependency to only apply for @2.28.1 or higher, you could rewrite the depends_on as follows:

I tried this but then again the concretizer fails:
```
$ spack spec atk
Input spec
--------------------------------
atk

Concretized
--------------------------------
==> Error: An unsatisfiable version constraint has been detected for spec:

    python@2.7.15%gcc@7.3.0+dbm~optimizations+pic+pythoncmd+shared~tk~ucs4 arch=linux-ubuntu16.04-x86_64 
        ^bzip2@1.0.6%gcc@7.3.0+shared arch=linux-ubuntu16.04-x86_64 
        ^gdbm@1.14.1%gcc@7.3.0 arch=linux-ubuntu16.04-x86_64 
            ^readline@7.0%gcc@7.3.0 arch=linux-ubuntu16.04-x86_64 
                ^ncurses@6.1%gcc@7.3.0~symlinks~termlib arch=linux-ubuntu16.04-x86_64 
                    ^pkgconf@1.4.2%gcc@7.3.0 arch=linux-ubuntu16.04-x86_64 
        ^openssl@1.0.2o%gcc@7.3.0+systemcerts arch=linux-ubuntu16.04-x86_64 
            ^zlib@1.2.11%gcc@7.3.0+optimize+pic+shared arch=linux-ubuntu16.04-x86_64 
        ^sqlite@3.23.1%gcc@7.3.0~functions arch=linux-ubuntu16.04-x86_64 


while trying to concretize the partial spec:

    atk@2.28.1%gcc@7.3.0 arch=linux-ubuntu16.04-x86_64 
        ^glib@2.56.2%gcc@7.3.0~libmount tracing= arch=linux-ubuntu16.04-x86_64 
            ^gettext@0.19.8.1%gcc@7.3.0+bzip2+curses+git~libunistring+libxml2+tar+xz arch=linux-ubuntu16.04-x86_64 
                ^bzip2@1.0.6%gcc@7.3.0+shared arch=linux-ubuntu16.04-x86_64 
                ^libxml2@2.9.8%gcc@7.3.0~python arch=linux-ubuntu16.04-x86_64 
                    ^pkgconf@1.4.2%gcc@7.3.0 arch=linux-ubuntu16.04-x86_64 
                    ^xz@5.2.4%gcc@7.3.0 arch=linux-ubuntu16.04-x86_64 
                    ^zlib@1.2.11%gcc@7.3.0+optimize+pic+shared arch=linux-ubuntu16.04-x86_64 
                ^ncurses@6.1%gcc@7.3.0~symlinks~termlib arch=linux-ubuntu16.04-x86_64 
                ^tar
            ^libffi@3.2.1%gcc@7.3.0 arch=linux-ubuntu16.04-x86_64 
            ^pcre+utf
            ^perl@5.26.2%gcc@7.3.0+cpanm+shared+threads arch=linux-ubuntu16.04-x86_64 
                ^gdbm@1.14.1%gcc@7.3.0 arch=linux-ubuntu16.04-x86_64 
                    ^readline@7.0%gcc@7.3.0 arch=linux-ubuntu16.04-x86_64 
            ^python@2.7.15%gcc@7.3.0+dbm~optimizations+pic+pythoncmd+shared~tk~ucs4 arch=linux-ubuntu16.04-x86_64 
                ^openssl@1.0.2o%gcc@7.3.0+systemcerts arch=linux-ubuntu16.04-x86_64 
                ^sqlite@3.23.1%gcc@7.3.0~functions arch=linux-ubuntu16.04-x86_64 
        ^gobject-introspection@1.49.2%gcc@7.3.0 arch=linux-ubuntu16.04-x86_64 
            ^bison@3.0.5%gcc@7.3.0 arch=linux-ubuntu16.04-x86_64 
                ^m4@1.4.18%gcc@7.3.0+sigsegv arch=linux-ubuntu16.04-x86_64 
                    ^libsigsegv
            ^cairo@1.14.12%gcc@7.3.0~X arch=linux-ubuntu16.04-x86_64 
                ^fontconfig@2.12.3%gcc@7.3.0 arch=linux-ubuntu16.04-x86_64 
                    ^font-util@1.3.1%gcc@7.3.0 arch=linux-ubuntu16.04-x86_64 
                        ^util-macros@1.19.1%gcc@7.3.0 arch=linux-ubuntu16.04-x86_64 
                    ^freetype@2.7.1%gcc@7.3.0 arch=linux-ubuntu16.04-x86_64 
                        ^libpng@1.6.34%gcc@7.3.0 arch=linux-ubuntu16.04-x86_64 
                    ^gperf
                ^pixman@0.34.0%gcc@7.3.0 arch=linux-ubuntu16.04-x86_64 
            ^flex@2.6.3%gcc@7.3.0+lex arch=linux-ubuntu16.04-x86_64 
                ^help2man@1.47.4%gcc@7.3.0 arch=linux-ubuntu16.04-x86_64 


atk requires python version 3:, but spec asked for 2.7.15
```

Looks like the same problem described in #2632

Reply:
Got it - I didn't think about that. That is another instance of #2632.